utils/imagenet_topk_accuracy_driver.py

```diff
@@ -0,0 +1,303 @@
+        processed_dir_path = os.path.join(processed_validation_images_dir, img_dir)
+
+        # Create a new subdirectory for the next label.
+        try:
+            os.makedirs(processed_dir_path)
+        except OSError:
+            sys.exit("New label subdirectory somehow already existed.")
+
+        # Transform and save all images in this label subdirectory.
+        for orig_img_filename in os.listdir(orig_dir_path):
+            orig_img_path = os.path.join(orig_dir_path, orig_img_filename)
+            if os.path.isfile(orig_img_path):
+                processed_img_path = os.path.join(processed_dir_path,
+                                                  orig_img_filename)
+                resize_and_save_image(orig_img_path, processed_img_path)
+
+## @returns a list of strings (of length equal to the batch size) which are
+# paths to images to do inference on. @param img_paths is the set of all image
+# paths, @param img_index is the next index to use in @param img_paths, and
+# @param tmp_dir_name is the location of where to save the images if necessary.
+def get_curr_img_paths(img_paths, img_index, tmp_dir_name):
+    curr_img_paths = []
+    for batch_idx in xrange(args.batch_size):
+        img_path = img_paths[img_index + batch_idx]
+        # If we are resizing the image then we are going to save it to a
+        # temp location to read in later for inference.
+        if args.resize_input_images:
+            # Save the new image to the tmp directory.
+            path_to_tmp_img = os.path.join(tmp_dir_name, "tmp_img" +
+                                           str(batch_idx) + ".png")
+            resize_and_save_image(img_path, path_to_tmp_img)
+            img_path = path_to_tmp_img
+
+        curr_img_paths.append(img_path)
+
+    return curr_img_paths
+
+## Verifies that the @param image_classifier_cmd is well formatted via
+## assertions.
+def verify_spawn_cmd(image_classifier_cmd):
+    split_cmd = image_classifier_cmd.split()
+
+    assert "image-classifier" in split_cmd[0], "image-classifier binary must be provided first; Perhaps another flag was spelled incorrectly."
+    assert "-" in split_cmd, "Streaming mode must be used"
+    assert "-topk=5" in split_cmd, "-topk=5 must be used"
+    assert any("-model_input_name=" in s for s in split_cmd), "image-classifier requires -model_input_name to be specified"
+    assert any("-m=" in s for s in split_cmd), "image-classifier requires -m to be specified"
+
+## Prints the Top-1 and Top-5 accuracy given @param total_image_count, @param
+## top1_count, and @param top5_count.
+def print_topk_accuracy(total_image_count, top1_count, top5_count):
+    top1_accuracy = float(top1_count) / float(total_image_count)
+    top5_accuracy = float(top5_count) / float(total_image_count)
+    print("\tTop-1 accuracy: " + "{0:.4f}".format(top1_accuracy))
+    print("\tTop-5 accuracy: " + "{0:.4f}".format(top5_accuracy))
+
+
+def main():
+    global args
```

Comment:
Can you avoid making `args` global, and pass the relevant bits down instead?

Reply:
Done!